indico/modules/events/registration/tasks.py

```diff
@@ -0,0 +1,61 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2022 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from collections import defaultdict
+
+from celery.schedules import crontab
+
+from indico.core.celery import celery
+from indico.core.db import db
+from indico.modules.events import Event
+from indico.modules.events.registration import logger
+from indico.modules.events.registration.models.form_fields import RegistrationFormField, RegistrationFormFieldData
+from indico.modules.events.registration.models.forms import RegistrationForm
+from indico.modules.events.registration.models.items import RegistrationFormItem
+from indico.modules.events.registration.models.registrations import RegistrationData
+from indico.util.date_time import now_utc
+
+
+def _group_by_regform(registration_data):
+    data_by_regform = defaultdict(list)
+    for data in registration_data:
+        data_by_regform[data.field_data.field.registration_form].append(data)
+    return data_by_regform
+
+
+@celery.periodic_task(name='registration_fields_retention_period', run_every=crontab(hour='*/24'))
```

Comment:
This is running every minute..  you need to pick a fixed minute/hour, e.g.:

```suggestion
@celery.periodic_task(name='registration_fields_retention_period', run_every=crontab(minute='0', hour='3'))
```